src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Helpers/ScaffolderBuilderExtensions.cs

```diff
@@ -0,0 +1,116 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using Microsoft.DotNet.Scaffolding.Helpers.General;
+using Microsoft.DotNet.Scaffolding.Helpers.Steps;
+using Microsoft.DotNet.Scaffolding.Helpers.Templates.DbContext;
+using Microsoft.DotNet.Scaffolding.TextTemplating;
+
+namespace Microsoft.DotNet.Scaffolding.Core.Builder;
+
+internal static class ScaffolderBuilderExtensions
+{
+    /// <summary>
+    /// Adds a new DbContext class 
+    /// </summary>
+    /// <param name="builder"></param>
+    /// <returns></returns>
+    public static IScaffoldBuilder WithDbContextStep(
+        this IScaffoldBuilder builder,
+        DbContextProperties? dbContextProperties = null)
+    {
+        builder = builder.WithStep<TextTemplatingStep>(config =>
+        {
+            var step = config.Step;
+            var context = config.Context;
+            if (dbContextProperties is null &&
+                context.Properties.TryGetValue("DbContextProperties", out var dbContextPropertiesObj) &&
+                dbContextPropertiesObj is DbContextProperties)
+            {
+                dbContextProperties = dbContextPropertiesObj as DbContextProperties;
+            }
+
+            if (dbContextProperties is null ||
+                string.IsNullOrEmpty(dbContextProperties.DbContextPath))
+            {
+                config.Step.SkipStep = true;
+                return;
+            }
+
+            GetDbContextTemplatingStepProperties(out var templatePath, out var templateType, out var templateModelName);
+            step.TemplatePath = templatePath;
+            step.TemplateType = templateType;
+            step.TemplateModelName = templateModelName;
+            step.TemplateModel = dbContextProperties;
+            step.OutputPath = dbContextProperties.DbContextPath;
+        });
+
+        return builder;
+    }
+
+    public static IScaffoldBuilder WithConnectionStringStep(
+        this IScaffoldBuilder builder,
+        string? baseProjectPath = null,
```

Comment:
Same question about `baseProjectPath` and `dbContextProperties` here as I had about `dbContextProperties` in the other extension method - doesn't appear to be used in any of the current calls.

Reply:
yep removed as well